lib/internal/modules/esm/resolve.js

```diff
@@ -850,18 +856,22 @@ function shouldBeTreatedAsRelativeOrAbsolutePath(specifier) {
  * @returns {URL}
  */
 function moduleResolve(specifier, base, conditions) {
```

Comment:
I would really prefer we leave this function as-is and treat it as `localModuleResolve`. Especially with the CJS fallback checks in its usage.

Reply:
i'm not sure we should have duplicated code that needs to be kept in-sync but agree this is slightly more complex

Reply:
If we have more higher-level functions then the top-level duplication shoudl only ever be a few lines. This function was always supposed to be a short top-level method.